lib/html/proofer/checkable.rb

```diff
@@ -107,11 +107,13 @@ def internal?
       def file_path
         return if path.nil?
 
+        path_dot_ext = path + @check.options[:assume_extension].to_s
+
         if path =~ %r{^/} # path relative to root
           base = File.directory?(@check.src) ? @check.src : File.dirname(@check.src)
-        elsif File.exist?(File.expand_path path, @check.src) # relative links, path is a file
+        elsif File.exist?(File.expand_path path, @check.src) || File.exist?(File.expand_path path_dot_ext, @check.src) # relative links, path is a file
```

Comment:
Mind adding the parenthesis here for both of the calls? Not sure why they were missing in the first place...